tests/auth/github_test.py

```diff
@@ -34,6 +34,13 @@ def test_guest_merge(monkeypatch):
     s = fc.sr_auth_state(isLoggedIn=True, method='guest')
     fc.sr_get('authGithubAuthorized', query={'state': state})
     fc.sr_auth_state(method='github', uid=guest_uid)
+    fc.sr_post(
```

Comment:
@robnagler This "fixed" the test now that complete_registration() is no longer in srunit. I'm still not sure I understand the order of operations when a user logs in so please double check that this makes sense.

Reply:
When the test was written, it was assumed that all UserRegistration records had display_name so needCompleteRegistration was set for guest users. When someone converted from guest to github, they already had a display_name. Now a guest will login with github and be asked to complete registration because needCompleteRegistration will be true and force an exception if they try to do anything else.

Reply:
Makes sense. Thanks.